sklearn/tree/_criterion.pyx

```diff
@@ -226,43 +238,40 @@ cdef class ClassificationCriterion(Criterion):
         self.weighted_n_right = 0.0
 
         # Count labels for each output
-        self.label_count_left = NULL
-        self.label_count_right = NULL
-        self.label_count_total = NULL
+        self.sum_total = NULL
+        self.sum_left = NULL
+        self.sum_right = NULL
         self.n_classes = NULL
 
         safe_realloc(&self.n_classes, n_outputs)
 
         cdef SIZE_t k = 0
-        cdef SIZE_t label_count_stride = 0
+        cdef SIZE_t sum_stride = 0
 
         # For each target, set the number of unique classes in that target,
         # and also compute the maximal stride of all targets
         for k in range(n_outputs):
             self.n_classes[k] = n_classes[k]
 
-            if n_classes[k] > label_count_stride:
-                label_count_stride = n_classes[k]
+            if n_classes[k] > sum_stride:
+                sum_stride = n_classes[k]
 
-        self.label_count_stride = label_count_stride
+        self.sum_stride = sum_stride
 
-        cdef SIZE_t n_elements = n_outputs * label_count_stride
-        self.label_count_left = <double*> calloc(n_elements, sizeof(double))
-        self.label_count_right = <double*> calloc(n_elements, sizeof(double))
-        self.label_count_total = <double*> calloc(n_elements, sizeof(double))
+        cdef SIZE_t n_elements = n_outputs * sum_stride
+        self.sum_total = <double*> calloc(n_elements, sizeof(double))
+        self.sum_left = <double*> calloc(n_elements, sizeof(double))
+        self.sum_right = <double*> calloc(n_elements, sizeof(double))
 
-        if (self.label_count_left == NULL or
-                self.label_count_right == NULL or
-                self.label_count_total == NULL):
+        if (self.sum_total == NULL or 
+            self.sum_left == NULL or
+            self.sum_right == NULL):
```

Comment:
Currently, we have
```
        if (self.sum_total == NULL or 
            self.sum_left == NULL or
            self.sum_right == NULL):
            raise MemoryError()
```

However, we can distinguish the indentation level between `self.sum_right == NULL):` and `raise MemoryError()`.
In that case, pep8 suggests to indent the code in this way

```
        if (self.sum_total == NULL or 
                self.sum_left == NULL or
                self.sum_right == NULL):
            raise MemoryError()
```